### PR TITLE
CASMCMS-7691: Fix testcase bugs in new IMS signing keys test

### PR DESCRIPTION
**Pull request by mharding-hpe**
## Summary and Scope

TLDR; This fixes two testcase bugs in the new IMS signing keys test. Without these fixes, this test always fails, which is bad since this test is part of the standard CSM health check procedure performed during installs.

In csm-1.2 a new subtest was added to the cmsdev tool to test IMS signing keys. When the test was added, no system was available with the new feature, so the test could not be successfully run before it was merged. As it turns out, the test had two bugs in it. This PR addresses those. Specifically:
* The test used the kubectl command to get information from a Kubernetes configmap. Something in the way it was doing that was causing issues with YAML parsing. I verified that the information in Kubernetes was correct, which led me to conclude that this was some golang subtlety that I was unable to detect. However, I also realized that we should not be using kubectl to get this information in the first place. Instead, we should be using the golang kubernetes API. I changed the test to use the API and the problem was resolved.
* The test was trying to run the podman command incorrectly. It wanted to pass an empty argument to the entrypoint parameter, but it was doing it shell script style (passing in two quotation marks). When calling the command from golang, it is sufficient just to pass an empty golang string for that argument. Once I made that change, the podman commands ran as they were intended to.

While originally investigating this issue I wanted to rule out that the golang version was not somehow causing a problem. I noticed that the Jenkins build output did not include the golang version being used, so I added a line in the RPM specfile to print the golang version at build time.

## Issues and Related PRs

This resolves CASMCMS-7691 (and whatever tickets have been opened that were duplicates of it).

## Testing

I tested the updated cmsdev tool on wasp, the only 1.2 system I had at hand. I verified that the test was failing without the fix, and passes with the fix.

## Risks and Mitigations

It's always possible that I inadvertently broke something else in trying to fix this, but I think it's unlikely. Given that my changes are confined to this subtest (and creating a new function that is called only by this subtest), it seems unlikely that any unexpected failures would extend beyond this subtest. And given that this subtest fails every time without these fixes, the worst case is that it still wouldn't work. But the changes I made are fairly simple in scope, so I think the risk of new problems being introduced is low.

It *is* possible that we may still see this test fail on some systems. I say this because we haven't been able to run it at all until I fixed these bugs. So it's possible the test has other bugs in it that we have not yet detected, which would only be exposed when it is run on different systems. Or it is possible that some systems will have actual issues with the IMS feature being tested, in which case the test failure would be desirable.

## Pull Request Checklist

- [N/A] Version number(s) incremented, if applicable
- [N/A] Copyrights updated
- [X] License file intact
- [X] Target branch correct
- [N/A] CHANGELOG.md updated
- [X] Testing is appropriate and complete, if applicable
- [N/A] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

File: cmsdev/internal/test/ims/signingkeys.go [hunk 1]
```diff
@@ -39,31 +39,33 @@ import (
...
-	cmdOut, err = cmd.CombinedOutput()
-	common.Debugf("OUT: %s", cmdOut)
-	if err == nil && len(cmdOut) == 0 {
-		err = fmt.Errorf("Command succeeded but gave no output")
-		return
-	} else if err != nil {
-		common.Errorf("Error getting Open SuSE Image from cray-configmap-ims-v2-image-create-kiwi-ng")
+
+	// Retrieve image_job_create.yaml.template data field
+	common.Debugf("Retrieve Data field 'imageFieldName' from ConfigMap", imageFieldName)
```
Comment (dlaine-hpe) [inline cmsdev/internal/test/ims/signingkeys.go:58]: Is this a way of formatting go strings I haven't seen, or should this be %s instead of 'imageFieldName' in the format string?

**Review by dlaine-hpe — APPROVED** [review]

File: cmsdev/internal/test/ims/signingkeys.go (hunk 1 above)
Comment (mharding-hpe) [reply cmsdev/internal/test/ims/signingkeys.go:58]: You are correct, sir, this is a mistake. Good catch!

Comment (mharding-hpe) [reply cmsdev/internal/test/ims/signingkeys.go]: Fixed!

File: cmsdev/internal/lib/k8s/k8s.go [hunk 2]
```diff
@@ -354,6 +354,30 @@ func GetNodeNames(params ...string) ([]string, error) {
...
+
+	cmlist, err := clientset.CoreV1().ConfigMaps(namespace).List(v1.ListOptions{})
+	if err != nil {
+		return
+	}
+
+	for _, cm = range cmlist.Items {
+		if cm.ObjectMeta.Name == name {
+			common.Debugf("Found Kubernetes ConfigMap in namespace %s with name %s", namespace, name)
+			return
```
Comment (ecozzi-hpe) [inline cmsdev/internal/lib/k8s/k8s.go:374]: Is this function just checking for the existence of the cm? Or, is it supposed to return the actual cm data? Am I missing something? I'm not a go person, by do you need to `return cm`?

**Review by rkleinman-hpe — APPROVED** [review]

File: cmsdev/internal/lib/k8s/k8s.go (hunk 2 above)
Comment (mharding-hpe) [reply cmsdev/internal/lib/k8s/k8s.go:374]: It's a golang thing -- this function does return **cm** when it finds it. 

In go you can name the return variables in the function statement. In this case, in the function statement I declared that the cm variable would hold the return value for the ConfigMap (and the err variable would hold the error value, if any). That's why all of the return statements have no arguments.

Comment (ecozzi-hpe) [reply cmsdev/internal/lib/k8s/k8s.go:374]: Got it. Thanks.

**Review by ecozzi-hpe — APPROVED** [review]

File: cmsdev/internal/lib/k8s/k8s.go (hunk 2 above)
Comment (mharding-hpe) [reply cmsdev/internal/lib/k8s/k8s.go:374]: Note that this type of syntax is optional -- you can declare a function and not name the return variables (you do still have to declare their types), and then use return statements in the more traditional way. It all comes down to what makes more sense for the given function.
